example/docker-compose/azure/docker-compose.yaml

```diff
@@ -12,7 +12,7 @@ services:
       - "3200:3200"  # tempo
     depends_on:
       - azure-cli
-      - azurite
+      - azurite      # TODO Tempo fails to start if azurite is not accepting connections yet, we should wait
```

Comment:
I'm not sure what's up here, this used to work in the past 🤷🏻  Tempo must have become too speedy 🏎

Tempo fails to start with a regular `docker-compose up -d` because azurite is refusing connections:
```
Attaching to azure_tempo_1
tempo_1                     | level=info ts=2021-08-13T09:28:29.146528089Z caller=main.go:190 msg="initialising OpenTracing tracer"
tempo_1                     | level=info ts=2021-08-13T09:28:29.147751797Z caller=main.go:109 msg="Starting Tempo" version="(version=grafana-8-f822caa-WIP, branch=grafana-8, revision=f822caa9)"
tempo_1                     | level=error ts=2021-08-13T09:28:29.151660339Z caller=main.go:112 msg="error running Tempo" err="failed to init module services error initialising module: store: failed to create store getting storage container: Get \"http://azurite:10000/devstoreaccount1/tempo?restype=container&timeout=61\": dial tcp 172.22.0.2:10000: connect: connection refused"
```
Running `docker-compose up -d` again fixes this. So it might not be worth fixing this.

Underlying issue: `depends_on` does not make this container wait until the given container is healthy, it only signals docker-compose to launch all of them together (i.e. if you run the service tempo, you must also run azurite). If we want to wait we have to add a wait script or whatever.
https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on
https://docs.docker.com/compose/startup-order/

Reply:
Yes, this is annoying.  Add a `restart:always` ?

Reply:
Nice, that works!